VogtObserver/Program.cs

```diff
@@ -8,33 +8,26 @@ class Program
         static void Main(string[] args)
         {
             RandomNumberGenerator generator = new RandomNumberGenerator();
-            Date dateTime = new Date();
-            Sum sum = new Sum();
+            Date dateTime = new Date(generator);
+            Sum sum = new Sum(generator);
 
             Run(generator, dateTime, sum);
         }
 
         public static void Run(RandomNumberGenerator generator, Date dateTime, Sum sum)
         {
-            generator.Add(dateTime);
-            generator.Add(sum);
 
-            for (int i = 1; i <=20 ; i++)
-            {
-                try
-                {
-                    Console.Clear();
+            dateTime = new Date(generator);
```

Comment:
Why create a new observer? (dateTime and sum)

youre passing those in from the Main() method already. no need to create ones by calling `new Date(generator)`. Thats wasting the memory you assigned on line 11 wher eyou create a `new Date(generator)` too. 

Btw, when you call `new Date(generator)`.. the new keyword allocates memory for that object.

If you don't use that object, and overwrite with yet another `new Date(generator)` you've now created Garbage that the program has to clean up (`Garbage Collector` im sure youve heard of)